### ISSUE TITLE
Let's choose a topic

### ISSUE THREAD

**Issue by pjurczynski** [pjurczynski/inmate_kata#1]
This time I would like to not only create a simple function that can do stuff but actually implement UI through TDD. I have a couple of suggestions:

Safe bet:
* simple calculator with an interface

Almost ambitious:
* again the polish notation, but we should finish it and implement an interface.

Ambitious:
* A parser for the algebraic notation in chess: https://en.wikipedia.org/wiki/Algebraic_notation_(chess) - I would like to evolve it to have an actual interface, maybe with a prompt for "next move"

**Comment by Greegko** [comment]
I would like to do this calculator polish notation, because it seems quite extensionable but still relatively small and focused. Also would start from scratch to discuss the initial problem and implementation as well.

**Comment by pjurczynski** [comment]
ok! Sounds terrific! I'll prepare better this time ;) https://en.wikipedia.org/wiki/Reverse_Polish_notation

**Comment by pjurczynski** [comment]
I read the article once again and I think I know what we did wrong the first time (apart from not reading the instruction first and going bare foot into the fire ;)).

First, split the problem in two areas
1. Binary operations
2. Unary operations

Next, tackle only one type of operations (we did that, but I think I got distracted by the fact we're not covering the other case and I got a thinking block). This might close ourselves from some optimal answers, but this feels easier when you don't have an idea how to solve a problem.

Let's do the same. Let's focus on binary calculations.

Another problem I had is I thought that TDD could magically find an answer to a problem by going step by step. But now I think you still need to have a plan in your head, and try to follow it step by step (I think you already said that but with different words).

Currently I have two ideas - binary tree, or a similar approach we had before, but this time working ;). I'll think more later.
